public/app/plugins/datasource/cloudwatch/__mocks__/CloudWatchDataSource.ts

```diff
@@ -82,6 +82,7 @@ export function setupMockedDataSource({
   datasource.api.getNamespaces = jest.fn().mockResolvedValue([]);
   datasource.api.getRegions = jest.fn().mockResolvedValue([]);
   datasource.api.getDimensionKeys = jest.fn().mockResolvedValue([]);
+  datasource.api.getMetrics = jest.fn().mockResolvedValue([]);
```

Comment:
I think it's because this hook is not calling the getDimensionValues on render - it's only called when a region, metric name or dimension key is changed.